server/base/src/main/java/org/apache/accumulo/server/rpc/HighlyAvailableServiceTMultiplexedProcessor.java

```diff
@@ -0,0 +1,204 @@
+   *         called {@link #registerProcessor(String, TProcessor) registerProcessor} during
+   *         initialization, right? :)
+   */
+  @Override
+  public void process(TProtocol iprot, TProtocol oprot) throws TException {
+    /*
+     * Use the actual underlying protocol (e.g. TBinaryProtocol) to read the message header. This
+     * pulls the message "off the wire", which we'll deal with at the end of this method.
+     */
+    TMessage message = iprot.readMessageBegin();
+
+    if (message.type != TMessageType.CALL && message.type != TMessageType.ONEWAY) {
+      throw new TProtocolException(TProtocolException.NOT_IMPLEMENTED,
+          "This should not have happened!?");
+    }
+
+    // If the service is upgrading, throw an exception
+    if (service.isUpgrading()) {
+      LOG.info("Service can not be accessed while it is upgrading.");
+      if (message.type == TMessageType.ONEWAY) {
+        return;
+      } else {
+        throw new ThriftNotActiveServiceException(service.getServiceName(),
+            "Service can not be accessed while it is upgrading");
+      }
+    }
+
+    // If the service is not active, throw an exception
+    if (!service.isActiveService()) {
+      LOG.info("Denying access to RPC service as this instance is not the active instance.");
+      if (message.type == TMessageType.ONEWAY) {
+        return;
+      } else {
+        throw new ThriftNotActiveServiceException(service.getServiceName(),
+            "Denying access to RPC service as this instance is not the active instance");
+      }
+    }
+
+    // Extract the service name
+    int index = message.name.indexOf(TMultiplexedProtocol.SEPARATOR);
+    if (index < 0) {
+      if (defaultProcessor != null) {
+        // Dispatch processing to the stored processor
+        defaultProcessor.process(new StoredMessageProtocol(iprot, message), oprot);
+        return;
+      }
+      throw new TProtocolException(TProtocolException.NOT_IMPLEMENTED,
+          "Service name not found in message name: " + message.name + ".  Did you "
+              + "forget to use a TMultiplexProtocol in your client?");
+    }
+
+    // Create a new TMessage, something that can be consumed by any TProtocol
+    String serviceName = message.name.substring(0, index);
+    TProcessor actualProcessor = SERVICE_PROCESSOR_MAP.get(serviceName);
+    if (actualProcessor == null) {
+      throw new TProtocolException(TProtocolException.NOT_IMPLEMENTED, "Service name not found: "
+          + serviceName + ".  Did you forget to call registerProcessor()?");
+    }
+
+    // Create a new TMessage, removing the service name
+    TMessage standardMessage = new TMessage(
+        message.name.substring(serviceName.length() + TMultiplexedProtocol.SEPARATOR.length()),
+        message.type, message.seqid);
+
+    // Dispatch processing to the stored processor
+    actualProcessor.process(new StoredMessageProtocol(iprot, standardMessage), oprot);
```

Comment:
See https://github.com/apache/accumulo/issues/5411#issuecomment-2734458636. If we used this custom TMultiplexedProcessor for all server types and wrapped this method call with a try/catch, then we could handle better all `oneway` RPC calls that happen to throw an exception.

Reply:
Nothing, it was just an idea that expands this current idea. I wanted to get feedback on the idea.